src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -34,7 +29,7 @@ data class WalletState(val addresses: Map<String, List<UnspentItem>>, val parent
         swapInParams = swapInParams, currentBlockHeight = currentBlockHeight, all = utxos,
     )
 
-    data class Utxo(val previousTx: Transaction, val outputIndex: Int, val blockHeight: Long) {
+    data class Utxo(val txId: TxId, val outputIndex: Int, val blockHeight: Long, val previousTx: Transaction) {
```

Comment:
why not use `previousTx.txId` here ?

Reply:
It's not strictly necessary, but it's a first step before making `previousTx` optional in a future PR.